automation/workflow/workflow_test.go

```diff
@@ -0,0 +1,165 @@
+package workflow
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/qri-io/qri/profile"
+)
+
+// func TestWorkflowAdvance(t *testing.T)
+// func TestWorkflowComplete(t *testing.T)
+
+func TestMemStoreIntegration(t *testing.T) {
+	store := NewMemStore()
+	AssertStore(t, store)
+}
+
+func AssertStore(t *testing.T, store Store) {
+	aliceDatasetID := "alice_dataset_id"
+	aliceProID := profile.ID("alice_pro_id")
+	aliceSeedStr := "alice workflow assert store seed string"
+	SetIDRand(strings.NewReader(aliceSeedStr))
+	aliceID := NewID()
+	aliceTestTrigger := NewTestTrigger(aliceID)
+	// aliceTestHook := NewMemHook()
+	alice := &Workflow{
+		ID:        aliceID,
+		DatasetID: aliceDatasetID,
+		OwnerID:   aliceProID,
+		Triggers:  []Trigger{aliceTestTrigger},
+		// Hooks:     []Hook{aliceTestHook},
+	}
+	// got, err := store.Create(aliceDatasetID, aliceProID, []Trigger{aliceTestTrigger}, []Hook{aliceTestHook})
+	SetIDRand(strings.NewReader(aliceSeedStr))
+	got, err := store.Create(aliceDatasetID, aliceProID, []Trigger{aliceTestTrigger})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if diff := cmp.Diff(alice, got, cmpopts.IgnoreFields(Workflow{}, "Created"), cmp.AllowUnexported(TestTrigger{})); diff != "" {
```

Comment:
I'm already mocking IDs, should I also be mocking time? Chose to just ignore the field for now, but it doesn't seem logical to be mocking some things and not others.